### PR TITLE
Match comments not immediately followed by a space

### PR DESCRIPTION
**Pull request by ralish**
Fixes Issue #19.

### CODE REVIEW

**Comment by ketanhwr** [comment]
@stephpy is there some issue with this PR? If it's fine can you merge this?

File: after/syntax/yaml.vim [hunk 1]
```diff
@@ -40,7 +40,7 @@ syn keyword yamlConstant NULL Null null NONE None none NIL Nil nil
 syn keyword yamlConstant TRUE True true YES Yes yes ON On on
 syn keyword yamlConstant FALSE False false NO No no OFF Off off
 
-syn match  yamlKey	"^\s*\zs[^ \t\"]\+\ze\s*:"
+syn match  yamlKey	"^\s*\zs[^ \t\"#]\+\ze\s*:"
 syn match  yamlKey	"^\s*-\s*\zs[^ \t\"\']\+\ze\s*:"
```
Comment (cbandy) [inline after/syntax/yaml.vim]: This line also needs to consider `#` to not match comments inside sequences of mappings. For example,

```yaml
- #foo: baz
```

```suggestion
 syn match  yamlKey	"^\s*-\s*\zs[^ \t\"\'#]\+\ze\s*:"
```

File: after/syntax/yaml.vim [hunk 2]
```diff
@@ -40,7 +40,7 @@ syn keyword yamlConstant NULL Null null NONE None none NIL Nil nil
 syn keyword yamlConstant TRUE True true YES Yes yes ON On on
 syn keyword yamlConstant FALSE False false NO No no OFF Off off
 
-syn match  yamlKey	"^\s*\zs[^ \t\"]\+\ze\s*:"
+syn match  yamlKey	"^\s*\zs[^ \t\"#]\+\ze\s*:"
```
Comment (cbandy) [inline after/syntax/yaml.vim]: It looks as though this and the following line should match. While you're here, do you mind including a `\'` to treat strings with colons as strings? For example,

```yaml
'foo: baz'
```

```suggestion
syn match  yamlKey	"^\s*\zs[^ \t\"\'#]\+\ze\s*:"
```

**Comment by ralish** [comment]
@cbandy Thanks for those, I've applied the changes :-)

**Comment by stephpy** [comment]
Thanks, sorry for delay ...

**Pull request merged**